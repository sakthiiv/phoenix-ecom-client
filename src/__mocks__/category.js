export default {
  create: category => {},
  update: category => {},
  delete: categoryId => {},
  getById: categoryId => {},
  getAll: jest.fn(() => {
    return Promise.resolve([
      {
        id: "5bc9a87d76e3833f73511a20",
        description: "Description for electronics goes here from Mock.",
        name: "Electronics",
        subCategory: [
          {
            id: null,
            description: "Hi Sub",
            name: "nameSub",
            subCategory: null,
            valid: null
          }
        ],
        valid: null
      }
    ]);
  })
};
