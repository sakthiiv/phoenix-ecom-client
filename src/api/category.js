import fetch from "../helpers/fetch";
const domainPath = "category";

export default {
  create: category => {
    return fetch.post(domainPath, category);
  },
  update: (categoryId, category) => {
    return fetch.put(domainPath + "/" + categoryId, category);
  },
  delete: categoryId => {
    return fetch.delete(domainPath + "/" + categoryId);
  },
  getById: categoryId => {
    return fetch.get(domainPath + "/" + categoryId);
  },
  getAll: () => {
    return fetch.get(domainPath);
  }
};
