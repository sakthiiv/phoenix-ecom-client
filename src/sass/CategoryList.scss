.cart-title-container {
    width: 100%;
}

.cart-title-left {
    width: 95%;
    display: inline-block;
    vertical-align: middle;
}

.cart-title-right {
    display: inline-block;
    vertical-align: middle;
}