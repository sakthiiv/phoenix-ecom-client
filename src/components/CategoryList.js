import React, { Component } from "react";
import category from "../api/category";
import { Card, CardText, CardBody, Row, Col } from "reactstrap";
import { Modal, Panel } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import UpdateCategory from "./UpdateCategory";
import DeleteItem from "./DeleteItem";

class CategoryList extends React.Component {
  constructor() {
    super();
    this.state = {
      categories: [],
      showUpdateModal: false,
      showDeleteModal: false,
      formOpen: "update",
      currentCategory: null
    };
  }

  async componentDidMount() {
    const categories = await category.getAll();
    this.setState({ categories });
  }

  close() {
    this.setState({ showUpdateModal: false, showDeleteModal: false });
  }
  open = (e, type, category) => {
    // alert(e.target.name);
    this.setState({ formOpen: type, currentCategory: category });
    if (type === "update") {
      this.setState({ showUpdateModal: true });
    } else {
      this.setState({ showDeleteModal: true });
    }
  };

  onDeleteClick = () => {
    category.delete(this.state.currentCategory.id).then(() => {
      category.getAll().then(categories => {
        this.setState({ categories });
      });
      this.setState({ showDeleteModal: false });
    });
  };

  onDeleteCancelClick = () => {
    this.setState({ showDeleteModal: false });
  };

  statusRegister = status => {
    if (status) {
      this.setState({ showUpdateModal: false, showDeleteModal: false });
      category.getAll().then(categories => {
        this.setState({ categories });
      });
    } else {
      this.setState({ showUpdateModal: true, showDeleteModal: true });
    }
  };

  render() {
    const { formOpen, currentCategory } = this.state;

    return (
      <div>
        {this.state.categories.map((c, i) => {
          return (
            <div className="list-container" key={i}>
              <Row>
                <Col sm="12">
                  <Card>
                    <CardBody>
                      <div className="cart-title-container">
                        <div className="cart-title-left">
                          <h3>{c.name}</h3>
                        </div>
                        <div
                          className="cart-title-right fa-override"
                          onClick={e => {
                            this.open(e, "update", c);
                          }}
                        >
                          <FontAwesomeIcon className="fa-lg" icon="edit" />
                        </div>
                        <div
                          className="cart-title-right fa-override"
                          onClick={e => {
                            this.open(e, "delete", c);
                          }}
                        >
                          <FontAwesomeIcon className="fa-lg" icon="times" />
                        </div>
                      </div>
                    </CardBody>
                    {/* <CardImg
                      top
                      width="100%"
                      src="https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180"
                      alt="Card image cap"
                    /> */}
                    <CardBody className="card-body">
                      <CardText>{c.description}</CardText>
                    </CardBody>
                  </Card>
                </Col>
              </Row>
            </div>
          );
        })}
        <Modal show={this.state.showUpdateModal} onHide={this.close.bind(this)}>
          <Modal.Body>
            <Panel>
              <UpdateCategory
                category={currentCategory}
                statusRegister={this.statusRegister}
              />
            </Panel>
          </Modal.Body>
        </Modal>
        <Modal show={this.state.showDeleteModal} onHide={this.close.bind(this)}>
          <Modal.Body>
            <Panel>
              <DeleteItem type={"Category"} />
            </Panel>
          </Modal.Body>
          <Modal.Footer>
            <input type="button" value="Confirm" onClick={this.onDeleteClick} />
            <input
              type="button"
              value="Cancel"
              onClick={this.onDeleteCancelClick}
            />
          </Modal.Footer>
        </Modal>
      </div>
    );
  }
}

export default CategoryList;
