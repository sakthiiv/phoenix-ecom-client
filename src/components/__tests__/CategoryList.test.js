import React from "react";
import { shallow } from "enzyme";
import CategoryList from "../CategoryList";
import category from "../../__mocks__/category";

jest.mock("../api/category", () => {
  const categoryMock = require("../__mocks__/category");
  return categoryMock;
});

describe("CategoryList", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<CategoryList />);
    expect(wrapper).toMatchSnapshot();
  });

  it("sets the state componentDidMount", async () => {
    const renderedComponent = await shallow(<CategoryList />);
    await renderedComponent.update();
    expect(renderedComponent.state("categories").length).toEqual(1);
  });

  it("sets the state componentDidMount on error", async () => {
    const spy = jest
      .spyOn(category, "getAll")
      .mockImplementation(() => Promise.resolve("Error fetching categories"));

    const renderedComponent = await shallow(<CategoryList />);
    await renderedComponent.update();
    expect(renderedComponent.state("categories")).toEqual(
      "Error fetching categories"
    );
  });
});
