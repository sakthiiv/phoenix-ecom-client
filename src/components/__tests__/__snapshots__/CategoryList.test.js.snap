// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList renders correctly 1`] = `
<div>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Uncontrolled(Panel)>
        <UpdateCategory
          category={null}
          statusRegister={[Function]}
        />
      </Uncontrolled(Panel)>
    </ModalBody>
  </Modal>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={false}
  >
    <ModalBody
      bsClass="modal-body"
      componentClass="div"
    >
      <Uncontrolled(Panel)>
        <DeleteItem
          type="Category"
        />
      </Uncontrolled(Panel)>
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <input
        onClick={[Function]}
        type="button"
        value="Confirm"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Cancel"
      />
    </ModalFooter>
  </Modal>
</div>
`;
